modules/p2/ding_user_consent/ding_user_consent.install

```diff
@@ -5,6 +5,24 @@
  * Setup the fields needed for ding list.
  */
 
+/**
+ * Implements hook_uninstall().
+ */
+function ding_user_consent_uninstall() {
+  $url = variable_get('ding_user_consent_loan_history_store_description_link', '');
+  if ($url) {
+    list(, $nid) = explode('/', $url);
+    if ($nid) {
+      node_delete($nid);
+    }
+  }
+
+  variable_get('user_consent_activate');
+  db_delete('variable')
+    ->condition('name', 'ding_user_consent_%', 'like')
+    ->execute();
```

Comment:
Hmmm - can we do better than this?

Reply:
Such as? There's no wildcard support in `variable_del()`.

Reply:
If these is a large but limited amount of known variables we could delete them one by one.

Reply:
The admin form constructs variable names by invoking a provider function and looping over the result, so we don't actually know what names might exist.